src/platform/datapath_win.c

```diff
@@ -0,0 +1,636 @@
+    _In_opt_ const CXPLAT_TCP_DATAPATH_CALLBACKS* TcpCallbacks,
+    _In_opt_ QUIC_EXECUTION_CONFIG* Config,
+    _Out_ CXPLAT_DATAPATH** NewDataPath
+    )
+{
+    QUIC_STATUS Status = QUIC_STATUS_SUCCESS;
+    if (NewDataPath == NULL) {
+        Status = QUIC_STATUS_INVALID_PARAMETER;
+        goto Error;
+    }
+
+    Status =
+        DataPathInitialize(
+            ClientRecvContextLength,
+            UdpCallbacks,
+            TcpCallbacks,
+            Config,
+            NewDataPath);
+    if (QUIC_FAILED(Status)) {
+        QuicTraceLogVerbose(
+            DatapathInitFail,
+            "[  dp] Failed to initialize datapath, status:%d", Status);
+        goto Error;
+    }
+
+    Status =
+        RawDataPathInitialize(
+            ClientRecvContextLength,
+            Config,
+            (*NewDataPath),
+            &((*NewDataPath)->RawDataPath));
+    if (QUIC_FAILED(Status)) {
+        QuicTraceLogVerbose(
+            RawDatapathInitFail,
+            "[ raw] Failed to initialize raw datapath, status:%d", Status);
+        Status = QUIC_STATUS_SUCCESS;
+        (*NewDataPath)->RawDataPath = NULL;
+    }
+
+Error:
+
+    return Status;
+}
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+CxPlatDataPathUninitialize(
+    _In_ CXPLAT_DATAPATH* Datapath
+    )
+{
+    if (Datapath->RawDataPath) {
+        RawDataPathUninitialize(Datapath->RawDataPath);
+    }
+    DataPathUninitialize(Datapath);
+}
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+CxPlatDataPathUpdateConfig(
+    _In_ CXPLAT_DATAPATH* Datapath,
+    _In_ QUIC_EXECUTION_CONFIG* Config
+    )
+{
+    DataPathUpdateConfig(Datapath, Config);
+    if (Datapath->RawDataPath) {
+        RawDataPathUpdateConfig(Datapath->RawDataPath, Config);
+    }
+}
+
+_IRQL_requires_max_(DISPATCH_LEVEL)
+uint32_t
+CxPlatDataPathGetSupportedFeatures(
+    _In_ CXPLAT_DATAPATH* Datapath
+    )
+{
+    if (Datapath->RawDataPath) {
+        return DataPathGetSupportedFeatures(Datapath) |
+               RawDataPathGetSupportedFeatures(Datapath->RawDataPath);
+    }
+    return DataPathGetSupportedFeatures(Datapath);
+}
+
+_IRQL_requires_max_(DISPATCH_LEVEL)
+BOOLEAN
+CxPlatDataPathIsPaddingPreferred(
+    _In_ CXPLAT_DATAPATH* Datapath,
+    _In_ CXPLAT_SEND_DATA* SendData
+    )
+{
+    if (DatapathType(SendData) == CXPLAT_DATAPATH_TYPE_USER) {
+        return DataPathIsPaddingPreferred(Datapath);
+    } else if (DatapathType(SendData) == CXPLAT_DATAPATH_TYPE_XDP) {
+        return RawDataPathIsPaddingPreferred(Datapath);
+    } else {
+        CXPLAT_DBG_ASSERT(FALSE);
+    }
+    return FALSE;
+}
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+_Success_(QUIC_SUCCEEDED(return))
+QUIC_STATUS
+CxPlatDataPathGetLocalAddresses(
+    _In_ CXPLAT_DATAPATH* Datapath,
+    _Outptr_ _At_(*Addresses, __drv_allocatesMem(Mem))
+        CXPLAT_ADAPTER_ADDRESS** Addresses,
+    _Out_ uint32_t* AddressesCount
+    )
+{
+    // TODO: XDP doesn't support, could be inlined here
```

Comment:
Let's go ahead and inline these two functions please.

Reply:
I noticed that this API is not used at all. inline normal datapath should not affect anything.